### ISSUE TITLE
Servers without Heart react on rolls

### ISSUE THREAD

**Issue by RedCon329** [chiyadev/MudaeFarm#196]
Does this work on servers without heart react during rolls?

example:
![image](https://user-images.githubusercontent.com/68579011/89210475-c01ab880-d5f2-11ea-9d51-850a20ffc1a8.png)

if so, how? thanks :))

**Comment by joshoea** [comment]
You need to invite a maid with $invite. Thats why the hearts arent there

**Comment by phosphene47** [comment]
MudaeFarm doesn't support this yet.

**Comment by KingJackFrost** [comment]
I saw this thread, and I wanted to inquire a related question without raising another thread: does MudaeFarm not support claiming both kakera and/or character reacts? If not, would it work theoretically if we went in and edited the heartEmojis array to include the custom emojis? Or is that not possible due to how custom emojis work?

**Comment by phosphene47** [comment]
> does MudaeFarm not support claiming both kakera and/or character reacts?

I'm not sure what exactly you're trying to ask. It's clearly explained that MudaeFarm supports both, unless you're asking if it supports claiming without the reaction on rolls (which is what this issue is about)? Then no, it doesn't at the moment.

> If not, would it work theoretically if we went in and edited the heartEmojis array to include the custom emojis? Or is that not possible due to how custom emojis work?

Custom emotes are not unicode so you cannot add it to the heart emoji list. In fact, when you enable the "custom emotes" option in the configuration, all MudaeFarm is doing is *not* checking for heart emojis

**Comment by KingJackFrost** [comment]
> I'm not sure what exactly you're trying to ask. It's clearly explained that MudaeFarm supports both, unless you're asking if it supports claiming without the reaction on rolls (which is what this issue is about)? Then no, it doesn't at the moment.

So from what you say, if the server has custom emojis, then toggling "enable_custom_emotes" to true should work, right? I attempted it for both character reacts, and the bot doesn't recognize any of the characters being rolled (as they are custom emojis). I do know the server is recognized, as the bot recognizes embeds from things such as checking harem ($mm).

> Custom emotes are not unicode so you cannot add it to the heart emoji list. In fact, when you enable the "custom emotes" option in the configuration, all MudaeFarm is doing is not checking for heart emojis

Right, so does this check mean it's always trying to react, or is it doing something else? See issue above.

**Comment by phosphene47** [comment]
> So from what you say, if the server has custom emojis, then toggling "enable_custom_emotes" to true should work, right? I attempted it for both character reacts, and the bot doesn't recognize any of the characters being rolled (as they are custom emojis).

Strange, it should be reacting to the first reaction attached to the roll message if `enable_custom_emotes` is enabled.

**Comment by KingJackFrost** [comment]
I can attempt to replicate this issue on a private server, but it occurred to me another issue. Is the bot not scalable as the server and number of players get larger?

I did look into your code, and I noticed the toggle is essentially a catch all to bypass checking for the heart emojis. Thus, I assumed it might be a scalability issue. In this particular configuration, there are over 100k active players, and about 25 active servers to roll in.  I thought it was a scalability issue, but when I only configured it for one channel, it still had the same issue.

I hope this info is helpful in debugging. I can try to see if I can get logs to help, should that work too.

**Comment by phosphene47** [comment]
I don't believe the size of servers would affect MudaeFarm's ability to recognize and claim messages. If it does, it's likely that the problem lies in the Disqord client library not MudaeFarm, because MudaeFarm has no special handling for large servers. All servers are handled equally.

Nonetheless, if you do think it's a scalability issue, can you try splitting your account using profiles and run a MudaeFarm instance for each of them?

**Comment by KingJackFrost** [comment]
Just ran different profiles that all use the same account; it looks like MudaeFarm does not recognize the rolls with custom reactions. I know it can't be a scalability issue (as it definitely still recognizes other embeds). I suspect custom emojis are the issue. I wish I could tell you why. 

For reference, here is what it looks like (custom emoji example)

![image](https://user-images.githubusercontent.com/39563259/91918108-7e2f7180-ec87-11ea-82ff-62998a62fa67.png)
